dashboard/app/models/concerns/text_to_speech.rb

```diff
@@ -154,42 +154,48 @@ def tts_should_update_short_instructions?
   end
 
   def tts_long_instructions_text
-    if I18n.locale == I18n.default_locale
-      # Instructions in contained levels are used as TTS instead of the
-      # instructions of the containing level
-      return tts_long_instructions_override ||
-        TextToSpeech.sanitize(tts_for_contained_level || long_instructions || "")
+    # Instructions content priority:
+    #
+    # 1. manual override if it exists (English only)
+    # 2. contained level content if it exists
+    # 3. instructions content
+    if tts_long_instructions_override && I18n.locale == I18n.default_locale
+      tts_long_instructions_override
+    elsif contained_level_text = tts_for_contained_level
+      TextToSpeech.sanitize(contained_level_text)
     else
-      TextToSpeech.sanitize(try(:localized_long_instructions) || "")
+      TextToSpeech.sanitize(try(:localized_long_instructions) || long_instructions || "")
     end
   end
 
   def tts_for_contained_level
-    all_instructions = []
-
-    contained_levels.each {|contained| all_instructions.push(contained_level_text(contained))}
-    all_instructions.empty? ? nil : all_instructions * "\n"
+    all_instructions = contained_levels.map do |contained|
+      contained_level_text(contained)
+    end
+    all_instructions.empty? ? nil : all_instructions.join("\n")
   end
 
   def contained_level_text(contained)
     # For multi questions, create a string for TTS of the markdown, question, and answers
     if contained.long_instructions.nil?
-      combined_text = contained.properties["markdown"].nil? ? "" : contained.properties["markdown"] + "\n"
+      combined_text = []
+      if contained.properties["markdown"]
+        combined_text << contained.localized_text(contained.properties["markdown"])
+      end
       if contained.properties["questions"]
         contained.properties["questions"].each do |question|
-          combined_text += question["text"] + "\n"
+          combined_text << contained.localized_text(question["text"])
```

Comment:
I think the intent of the "\n" was to make sure that the audio is generated with a pause between the multiple answers/questions. I think adding the "\n" to line 195 won't add the "audio whitespace" between the options?

Reply:
I believe it will; before, we were assembling the string by appending "substring plus newline"; now, we're assembling the string by gathering the newlines into an array and joining them. The result should still be newlines between each substring.

Note in particular that the [existing tests](https://github.com/code-dot-org/code-dot-org/pull/29340/files#diff-ac3fafd71d55a368136f8f275923f521R103) did not change